File: code/java/ch19/practice/Warmup04CountComponents.java
package ch19.practice;

import java.util.*;

/**
 * Warmup 4: Count Connected Components
 * Chapter 19: Graphs I — Exploring Networks
 *
 * PROBLEM: Count the number of connected components in an undirected graph.
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Warmup04CountComponents {
    public static int solve(int n, int[][] edges) {
        // TODO: Replace this with your solution
        return 0;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt(), m = sc.nextInt();
        int[][] edges = new int[m][2];
        for (int i = 0; i < m; i++) { edges[i][0] = sc.nextInt(); edges[i][1] = sc.nextInt(); }
        System.out.println(solve(n, edges));
        sc.close();
    }
}

File: code/cpp/ch19/practice/warmup_04_count_components.cpp
/*
 * Warmup 4: Count Connected Components
 * Chapter 19: Graphs I — Exploring Networks
 *
 * PROBLEM: Count connected components in an undirected graph.
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
#include <iostream>
#include <queue>
#include <vector>
using namespace std;

int solve(int n, vector<vector<int>>& edges) {
    // TODO: Replace this with your solution
    return 0;
}

int main() {
    int n, m; cin >> n >> m;
    vector<vector<int>> edges(m, vector<int>(2));
    for (int i = 0; i < m; i++) cin >> edges[i][0] >> edges[i][1];
    cout << solve(n, edges) << endl;
    return 0;
}
